immuni_exposure_ingestion/helpers/risk_level.py

```diff
@@ -39,4 +41,9 @@ def extract_keys_with_risk_level_from_upload(upload: Upload) -> Iterable[Tempora
     for key in keys_at_risk:
         key.transmission_risk_level = TransmissionRiskLevel.highest
 
-    return keys_at_risk
+    # Also remove any keys that might still be valid
+    return (
+        [key for key in keys_at_risk if key.created_at.date() < date.today()]
```

Comment:
Can you please explain the rationale for this change? (either in the commit message or in the PR description)

This comparison seems to not take into account the `rolling_period` of the key. Based on current settings (period of `144`) keys would normally be active for 24h. A date comparison here will potentially drop keys which are actually expired because their `rolling_start_interval_number` is >24h ago. 

PS: Android's `getTemporaryExposureKeyHistory` specifies that no active key would be returned anyway, while Apple's documentation leaves this unspecified, so 👍 for implementing this additional check.

Reply:
An additional consideration on the effect of this change. 

The `process_uploads` task currently processes `Upload`s "atomically": after a call to `extract_keys_with_risk_level_from_upload` the `Upload` is flagged as published and thus excluded from subsequent task runs. If we change this method to drop active keys, the `Upload` will be regardless marked as published and thus the active key would never be processed again even when it later becomes inactive.

Reply:
@xelhark glad to see you [implemented](https://github.com/immuni-app/immuni-backend-common/pull/25) [the change](https://github.com/immuni-app/immuni-backend-exposure-ingestion/commit/4c79b274ec19fc2466a35ed37d5c37dd5f16984f) related to the rolling period. 👍

I see my comment above about the processing of discarded TEKs [translated](https://github.com/immuni-app/immuni-backend-exposure-ingestion/pull/24#discussion_r437482859) into a TODO, but I'd still appreciate your (or reviewers') thoughts on that topic.

In general, for the sake of encouraging a more open collaboration with external contributors, would be great to hear back in a comment before it gets resolved 🙏

Reply:
Hi!

I'm sorry I didn't answer, and you're completely right 🙏 I'll make sure to answer in comments before implementing.

The processing of discarded TEKs did translate into a TODO. We thought this would be a bigger change rather than simply discarding possibly active TEKs, and we need a bit more coordination in order to fix that.

About your other comment about the rolling period I also think you were right. Currently TEKs rolling periods are basically hardcoded at 144 (we also make sure that no key has a different value). This is due to some recommendations from Apple / Google, so the previous implementation would have worked anyways at the current state, but I agree this change makes the code both more robust and more readable.

Thanks a lot for your contribution! 👏